src/overlays/actors/ovl_Bg_Haka_Trap/z_bg_haka_trap.c

```diff
@@ -40,36 +41,490 @@ const ActorInit Bg_Haka_Trap_InitVars = {
+        }
+
+        Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+    }
+
+    func_800F89E8(&this->unk_16C);
+}
+
+void func_8087FFC0(BgHakaTrap* this, GlobalContext* globalCtx) {
+    f32 cosine;
+    Vec3f sp28;
+    f32 sine;
+    f32 zNonNegative;
+    Player* player = PLAYER;
+
+    func_8002DBD0(&this->dyna.actor, &sp28, &player->actor.posRot.pos);
+
+    sine = Math_Sins(this->dyna.actor.shape.rot.y);
+    cosine = Math_Coss(this->dyna.actor.shape.rot.y);
+    if (this->dyna.actor.params == TRAP_GUILLOTINE_SLOW) {
+        sp28.x = CLAMP(sp28.x, -50.0f, 50.0f);
+        zNonNegative = (sp28.z >= 0.0f) ? 1.0f : -1.0f;
+        sp28.z = zNonNegative * -15.0f;
+    } else {
+        sp28.x = -CLAMP(sp28.x, -162.0f, 162.0f);
+        zNonNegative = (sp28.z >= 0.0f) ? 1.0f : -1.0f;
+        sp28.z = zNonNegative * 15.0f;
+    }
+
+    this->colliderCylinder.dim.pos.x = (s16)((this->dyna.actor.posRot.pos.x + (sp28.x * cosine)) + (sp28.z * sine));
+    this->colliderCylinder.dim.pos.z = (s16)((this->dyna.actor.posRot.pos.z + (sp28.x * sine)) + (sp28.z * cosine));
+}
+
+void func_808801B8(BgHakaTrap* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 temp = 4;
+
+    if ((D_80880F30 == 0) && (func_8008E988(globalCtx) == 0)) {
+        if (!Math_ApproxF(&this->dyna.actor.posRot.pos.x, this->dyna.actor.initPosRot.pos.x, 0.5f)) {
+            func_8002F974(&this->dyna.actor, 0x2058);
+        } else if (this->dyna.actor.params == TRAP_SPIKED_WALL) {
+            D_80881018 |= 1;
+        } else if (this->dyna.actor.params == TRAP_SPIKED_WALL_2) {
+            D_80881018 |= 2;
+        }
+    }
+
+    func_8087FFC0(this, globalCtx);
+
+    if ((this->colliderSpikes.base.acFlags & 2) != 0) {
+        this->timer = 20;
+        D_80880F30 = 1;
+        this->actionFunc = func_808802D8;
+    } else if (D_80881018 == 3) {
+        D_80881018 = temp;
+        player->actor.bgCheckFlags |= 0x100;
+    }
+}
+
+void func_808802D8(BgHakaTrap* this, GlobalContext* globalCtx) {
+    Vec3f vector;
+    f32 xScale;
+    s32 i;
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    func_8002F974(&this->dyna.actor, 0x205B);
+
+    for (i = 0; i < 2; i++) {
+        f32 rand = Math_Rand_ZeroOne();
+
+        xScale = (this->dyna.actor.params == TRAP_SPIKED_WALL) ? -30.0f : 30.0f;
+
+        vector.x = xScale * rand + this->dyna.actor.posRot.pos.x;
+        vector.y = Math_Rand_ZeroOne() * 10.0f + this->dyna.actor.posRot.pos.y + 30.0f;
+        vector.z = Math_Rand_CenteredFloat(320.0f) + this->dyna.actor.posRot.pos.z;
+
+        func_8002A6B8(globalCtx, &vector, &D_8088101C, &D_8088101C, 0x82, 0x14, 0xFF, 0xFF, 0x96, 0xAA, 0xFF, 0, 0, 1,
+                      9, 0);
+    }
+
+    if (this->timer == 0) {
+        D_80880F30 = 0;
+        Actor_Kill(&this->dyna.actor);
+    }
+}
+
+void func_80880484(BgHakaTrap* this, GlobalContext* globalCtx) {
+    s32 sp24;
+    s32 timer;
+
+    if (this->unk_16A) {
```

Comment:
It's not obvious since the functions aren't named yet, but unk_16A is actually used as a boolean to tell if the guillotine is slow or fast in the `actionFunc`s for `HAKA_TRAP_GUILLOTINE_SLOW` and `HAKA_TRAP_GUILLOTINE_FAST`. 
In the `actionFunc`s for `HAKA_SPIKED_BOX`, unk_16A is used as an `s16` to store position data.

How should I make this clearer, maybe a comment or try using a union for unk_16A?